### PR TITLE
Support predeploying custom accounts

### LINKED ISSUE #285: Feature Request: Choose Account Contract used for Preloaded Accounts
Hey team. As we go through more upgrades and Account Contracts evolve, it's become evident that allowing us to choose which Account contract implementation used for the preloaded accounts would be beneficial (or at least the version of OZ used).

For example, now Pathfinder doesn't support Accounts post v0.10.0 due to this issues https://github.com/eqlabs/pathfinder/issues/632

Our testing therefore needs to be done with older accounts, but because we can't choose the Account used with the devnet, some friction is introduced.

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes

- Close #285 

## Development related changes

- Move DevnetBackgroundProc from `test/test_dump.py` to `test/util.py`
- Introduce `ContractClassWrapper`, a class that holds a `ContractClass` and its hash

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/lint.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: starknet_devnet/accounts.py [hunk 1]
```diff
@@ -20,6 +21,8 @@ def __init__(self, starknet_wrapper):
         self.__n_accounts = starknet_wrapper.config.accounts
         self.__initial_balance = starknet_wrapper.config.initial_balance
```
Comment (mikiw) [inline starknet_devnet/accounts.py:23]: There is a weird empty line here.

File: test/custom_account.json [hunk 2]
```diff
@@ -0,0 +1 @@
...":"cast([fp + (-5)] + 3, felt*)"}],"type":"reference"}},"main_scope":"__main__","prime":"0x800000000000011000000000000000000000000000000000000000000000001","reference_manager":{"references":[{"ap_tracking_data":{"group":1,"offset":0},"pc":3,"value":"[cast(fp + (-3), felt*)]"},{"ap_tracking_data":{"group":1,"offset":3},"pc":10,"value":"[cast(ap, felt*)]"},{"ap_tracking_data":{"group":2,"offset":0},"pc":18,"value":"[cast(fp + (-7), felt**)]"},{"ap_tracking_data":{"group":3,"offset":0},"pc":30,"value":"[cast(fp + (-3), felt**)]"},{"ap_tracking_data":{"group":4,"offset":0},"pc":37,"value":"[cast(fp + (-3), felt**)]"},{"ap_tracking_data":{"group":5,"offset":0},"pc":44,"value":"[cast(fp + (-4), felt**)]"},{"ap_tracking_data":{"group":6,"offset":0},"pc":52,"value":"[cast(fp + (-5), felt**)]"},{"ap_tracking_data":{"group":7,"offset":0},"pc":60,"value":"[cast(fp + (-3), felt**)]"},{"ap_tracking_data":{"group":8,"offset":0},"pc":67,"value":"[cast(fp + (-4), felt*)]"},{"ap_tracking_data":{"group":8,"offset":0},"pc":67,"value":"[cast(fp + (-3), felt*)]"},{"ap_tracking_data":{"group":8,"offset":0},"pc":67,"value":"[cast(fp + (-7), starkware.cairo.common.cairo_builtins.SignatureBuiltin**)]"}]}}}
```
Comment (mikiw) [inline test/custom_account.json:1]: Is there any reason for 1 line JSON file? Can we format them?

File: test/custom_account_missing_method.json [hunk 3]
```diff
@@ -0,0 +1 @@
...":"cast([fp + (-5)] + 3, felt*)"}],"type":"reference"}},"main_scope":"__main__","prime":"0x800000000000011000000000000000000000000000000000000000000000001","reference_manager":{"references":[{"ap_tracking_data":{"group":1,"offset":0},"pc":3,"value":"[cast(fp + (-3), felt*)]"},{"ap_tracking_data":{"group":1,"offset":3},"pc":10,"value":"[cast(ap, felt*)]"},{"ap_tracking_data":{"group":2,"offset":0},"pc":18,"value":"[cast(fp + (-7), felt**)]"},{"ap_tracking_data":{"group":3,"offset":0},"pc":30,"value":"[cast(fp + (-3), felt**)]"},{"ap_tracking_data":{"group":4,"offset":0},"pc":37,"value":"[cast(fp + (-3), felt**)]"},{"ap_tracking_data":{"group":5,"offset":0},"pc":44,"value":"[cast(fp + (-4), felt**)]"},{"ap_tracking_data":{"group":6,"offset":0},"pc":52,"value":"[cast(fp + (-5), felt**)]"},{"ap_tracking_data":{"group":7,"offset":0},"pc":60,"value":"[cast(fp + (-3), felt**)]"},{"ap_tracking_data":{"group":8,"offset":0},"pc":67,"value":"[cast(fp + (-4), felt*)]"},{"ap_tracking_data":{"group":8,"offset":0},"pc":67,"value":"[cast(fp + (-3), felt*)]"},{"ap_tracking_data":{"group":8,"offset":0},"pc":67,"value":"[cast(fp + (-7), starkware.cairo.common.cairo_builtins.SignatureBuiltin**)]"}]}}}
```
[the remark this answers is not in the document]
Comment (FabijanC) [reply test/custom_account_missing_method.json:1]: Takes up significantly less space. And if we ever want to read it, prettifying can be done very easily (one keyboard combination in vscode).

File: starknet_devnet/accounts.py (hunk 1 above)
Comment (FabijanC) [reply starknet_devnet/accounts.py]: Addressed in https://github.com/Shard-Labs/starknet-devnet/pull/294/commits/e2b2e1ff522c91f9449b41c8209c652af6563155

**Review by mikiw — APPROVED** [review]

**Pull request merged**